src/textual/_animator.py

```diff
@@ -368,3 +396,7 @@ def _get_time(self) -> float:
     async def wait_for_idle(self) -> None:
         """Wait for any animations to complete."""
         await self._idle_event.wait()
+
+    async def wait_for_fully_idle(self) -> None:
```

Comment:
I would rename this. Suggest `wait_until_complete`. Otherwise its not clear what idle means. I would take idle to mean "doing nothing right now".

Reply:
Fair point. How about `wait_until_done`?

Reply:
We use the term `complete` in reference to animations, so I think it makes sense to stick to that.

Reply:
Sorted in d91a10b